File: src/JS/insertionSort.js
function insertion_Sort(a)
{

    var n = a.length;
    var x=a;
    for (var i = 1; i < n; i++){
        var key = x[i];
        var j = i-1;

        while (j>=0 && x[j] > key){
            x[j+1] = x[j]; 
            j = j-1;
        }
        x[j+1] = key
    }

    return x;
}

File: src/Go lang/insertion-sort.go
package main

import "fmt"

func main() {
	nums := []int{1, 9, 4, 6, 3, 8, 2, 5, 7}
	fmt.Println(insertionSort(nums))
}

func insertionSort(a []int) []int {
	for i := 1; i < len(a); i++ {
		for j := 0; j < i; j++ {
			if a[i-j-1] > a[i-j] {
				a[i-j-1], a[i-j] = a[i-j], a[i-j-1]
			} else {
				break
			}
		}
	}
	return a
}
